src/api/routes/company.read.ts

```diff
@@ -314,143 +175,7 @@ export async function companyReadRoutes(app: FastifyInstance) {
-                        unknown: true,
-                        unit: true,
-                        metadata,
-                      },
-                    },
-                    scope3: {
-                      select: {
-                        statedTotalEmissions: {
-                          select: {
-                            total: true,
-                            unit: true,
-                            metadata,
-                          },
-                        },
-                        categories: {
-                          select: {
-                            category: true,
-                            total: true,
-                            unit: true,
-                            metadata,
-                          },
-                          orderBy: {
-                            category: 'asc',
-                          },
-                        },
-                        metadata,
-                      },
-                    },
-                    biogenicEmissions: {
-                      select: {
-                        total: true,
-                        unit: true,
-                        metadata,
-                      },
-                    },
-                    scope1And2: {
-                      select: {
-                        total: true,
-                        unit: true,
-                        metadata,
-                      },
-                    },
-                    statedTotalEmissions: {
-                      select: {
-                        total: true,
-                        unit: true,
-                        metadata,
-                      },
-                    },
-                  },
-                },
-                metadata,
-              },
-              orderBy: {
-                startDate: 'desc',
-              },
-            },
-            industry: {
-              select: {
-                industryGics: {
-                  select: {
-                    sectorCode: true,
-                    groupCode: true,
-                    industryCode: true,
-                    subIndustryCode: true,
-                  },
-                },
-                metadata,
-              },
-            },
-            goals: {
-              select: {
-                id: true,
-                description: true,
-                year: true,
-                baseYear: true,
-                target: true,
-                metadata,
-              },
-              orderBy: {
-                year: 'desc',
-              },
-            },
-            initiatives: {
-              select: {
-                id: true,
-                title: true,
-                description: true,
-                year: true,
-                scope: true,
-                metadata,
-              },
-              orderBy: {
-                year: 'desc',
-              },
-            },
-          },
+          ...detailedCompanyArgs,
```

Comment:
This works as expected, though it would be better to always spread before setting custom properties. In this case the `where` clause is set before we spread the rest of the args, which (in this case) don't include anything that would overwrite the `where` clause defined in the API endpoint.

It would be good to keep all spread operators at the start of each object to avoid ambiguous bugs.